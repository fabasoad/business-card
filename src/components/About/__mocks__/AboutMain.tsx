import * as React from 'react'

export default function AboutMain() {
  return (
    <div data-testid="AboutMain"></div>
  )
}
