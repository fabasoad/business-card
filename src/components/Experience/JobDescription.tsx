import * as React from 'react'
import { Trans, useTranslation } from 'react-i18next'

interface JobDescriptionProps {
  count: number
  i18nKeyPrefix: string
  withTitle?: boolean
}

export default function JobDescription(props: JobDescriptionProps) {
  const { t } = useTranslation()
  return (
    <>
      {props.withTitle && <h5>{t('business-card-experience-description-title')}</h5>}
      {Array(props.count).fill(props.i18nKeyPrefix)
        .map((p, i) => `${p}-${i + 1}`)
        .map((key) => (<Trans key={key} parent="div" t={t}>{key}</Trans>))}
    </>
  )
}

JobDescription.defaultProps = {
  withTitle: true
}
