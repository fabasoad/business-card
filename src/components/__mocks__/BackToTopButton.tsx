import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
