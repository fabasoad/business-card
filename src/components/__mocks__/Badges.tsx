import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
