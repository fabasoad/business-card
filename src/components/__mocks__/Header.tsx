import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
