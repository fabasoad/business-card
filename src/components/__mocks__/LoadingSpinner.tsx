import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
