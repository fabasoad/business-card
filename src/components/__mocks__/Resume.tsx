import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
