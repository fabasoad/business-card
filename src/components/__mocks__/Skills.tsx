import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
