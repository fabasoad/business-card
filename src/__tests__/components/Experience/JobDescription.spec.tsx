/// <reference types="jest" />
import * as React from 'react'
import { shallow } from 'enzyme'
import JobDescription from '../../../components/Experience/JobDescription'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock

beforeEach(() => {
  tMock = useTranslation().t
})

test('should render JobDescription without title correctly', () => {
  const prefix = 'some-prefix'
  const count = 2
  const wrapper = shallow(<JobDescription
    count={count}
    i18nKeyPrefix={prefix}
    withTitle={false}
  />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).not.toBeCalledWith('business-card-experience-description-title')
  for (let i = 0; i < count; i++) {
    expect(wrapper.childAt(i).childAt(0).text()).toBe(`${prefix}-${i + 1}`)
  }
})

test('should render JobDescription with title correctly', () => {
  const prefix = 'some-prefix'
  const count = 3
  const wrapper = shallow(<JobDescription
    count={count}
    i18nKeyPrefix={prefix}
  />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toBeCalledWith('business-card-experience-description-title')
  for (let i = 1; i < count; i++) {
    expect(wrapper.childAt(i).childAt(0).text()).toBe(`${prefix}-${i}`)
  }
})
