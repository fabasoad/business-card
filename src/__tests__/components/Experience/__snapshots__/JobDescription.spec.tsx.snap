// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render JobDescription with title correctly 1`] = `
<Fragment>
  <h5>
    business-card-experience-description-title
  </h5>
  <Trans
    key="some-prefix-1"
    parent="div"
    t={
      [MockFunction] {
        "calls": Array [
          Array [
            "business-card-experience-description-title",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": "business-card-experience-description-title",
          },
        ],
      }
    }
  >
    some-prefix-1
  </Trans>
  <Trans
    key="some-prefix-2"
    parent="div"
    t={
      [MockFunction] {
        "calls": Array [
          Array [
            "business-card-experience-description-title",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": "business-card-experience-description-title",
          },
        ],
      }
    }
  >
    some-prefix-2
  </Trans>
  <Trans
    key="some-prefix-3"
    parent="div"
    t={
      [MockFunction] {
        "calls": Array [
          Array [
            "business-card-experience-description-title",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": "business-card-experience-description-title",
          },
        ],
      }
    }
  >
    some-prefix-3
  </Trans>
</Fragment>
`;

exports[`should render JobDescription without title correctly 1`] = `
<Fragment>
  <Trans
    key="some-prefix-1"
    parent="div"
    t={[MockFunction]}
  >
    some-prefix-1
  </Trans>
  <Trans
    key="some-prefix-2"
    parent="div"
    t={[MockFunction]}
  >
    some-prefix-2
  </Trans>
</Fragment>
`;
